exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinMemoryCalculator.java

```diff
@@ -154,6 +151,9 @@ void initialize(boolean firstCycle,
     long getInMemorySize();
   }
 
+  interface HashJoinSpillControl {
+    boolean shouldSpill(VectorContainer currentVectorContainer);
```

Comment:
The argument currentVectorContainer shouldn't be required. So the signature should be

```
boolean shouldSpill();
```

For details about why we don't need the arg see my comment on HashJoinSpillControl

Reply:
The `currentVectorContainer` is used to asses the size of the next batch, which may vary as the incoming are read. If we make a (reasonable) assumption that there would be very little variation (as these are join keys), then that size can be given to the constructor and not passed as an argument to `shouldSpill()`.

Reply:
Removed this argument (i.e. restored original signature). See below.